Mathlib/Data/List/Chain.lean

```diff
@@ -337,9 +337,10 @@ theorem Chain'.induction (p : α → Prop) (l : List α) (h : Chain' r l)
   unfold Chain' at h
   split at h
   · simp
-  · simp_all only [ne_eq, not_false_eq_true, head_cons, true_implies, mem_cons, forall_eq_or_imp,
-      true_and]
-    exact h.induction p _ carries initial
+  · simp_all only [ne_eq, head_cons, mem_cons, forall_eq_or_imp]
+    constructor
+    exact initial (by simp)
+    exact h.induction p _ carries (initial (by simp))
```

Comment:
I don't understand why this proof repair was needed.

Reply:
Adding `reduceCtorEq` makes the original proof go through again.